frame/referenda/src/types.rs

```diff
@@ -256,6 +256,19 @@ impl<
 			Killed(_) => Ok(None),
 		}
 	}
+
+	/// Take the Submission Deposit from `self`, if there is one and it's in a valid state to be
+	/// taken. Returns an `Err` if `self` is not in a valid state for the Submission Deposit to be
+	/// refunded.
+	pub fn take_submission_deposit(&mut self) -> Result<Option<Deposit<AccountId, Balance>>, ()> {
```

Comment:
Return a proper `DispatchError` here? instead of doing `map_err` outside?

Reply:
This is a type method, not dependent from any dispatch/call concepts. I would better keep it this way